platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,194 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.common;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.google.common.collect.ImmutableSet;
+import ddf.security.samlp.SamlProtocol.Binding;
+import ddf.security.samlp.impl.EntityInformation;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class SPMetadataParserTest {
+  private static List<String> spMetadata;
+
+  private static Set<Binding> bindingSet;
+
+  private static final Map EMPTY_MAP = Collections.emptyMap();
+
+  private static String entityID2;
+  private static EntityInformation testInfo2;
+
+  private static String entityID;
+  private static EntityInformation testInfo;
+
+  private static final String FILE_PATH =
+      Paths.get("src", "test", "resources", "SPMetadata").toString();
+  private static final String FILE_PATH_2 = FILE_PATH + "2";
+  private static final String CERTIFICATE_PATH =
+      Paths.get("src", "test", "resources", "Certificate.cert").toString();
+
+  private static final String ENTITY_ID = "https://localhost:8993/services/saml/test";
+  private static final String ENTITY_ID_2 =
+      "http://localhost:80/spring-security-saml2-sample/saml/metadata";
+
+  private static final String REDIRECT_LOGOUT_URL =
+      "https://localhost:8993/services/saml/logout/redirect";
+  private static final String POST_LOGOUT_URL = "https://localhost:8993/services/saml/logout/post";
+  private static final String REDIRECT_ACS_URL =
+      "https://localhost:8993/services/saml/sso/redirect";
+  private static final String POST_ACS_URL = "https://localhost:8993/services/saml/sso/post";
+  private static final String SOAP_ACS_URL = "https://localhost:8993/services/saml/sso/soap";
+  private static String CERTIFICATE;
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    // read Certificate file into certificate
+    try (Stream<String> lines = Files.lines(Paths.get(CERTIFICATE_PATH))) {
+      CERTIFICATE = lines.collect(Collectors.joining(System.lineSeparator()));
+    }
```

Comment:
❓ Should we go through the class/classloader instead?

Reply:
See above comment